cmake/protobuf-configure-target.cmake

```diff
@@ -7,7 +7,7 @@
 
 # Refactors configuration options set on all Protobuf targets
 function(protobuf_configure_target target)
-    target_compile_features("${target}" PUBLIC cxx_std_17)
+    target_compile_features("${target}" PRIVATE cxx_std_17)
```

Comment:
I actually think this needs to be PUBLIC for now.  We make no effort to hide C++17 features in our internals, so our headers almost for sure require C++17.  This means anyone who uses protobuf libraries should have this feature too.  We could remove this from binaries, but I'm not sure private vs public even really matters there

Reply:
Ah nvm I saw your other comment, good point

Reply:
Fine, revert it.